src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -262,6 +267,28 @@ public InnReachResponseDTO returnUncirculated(String trackingId, String centralC
     }
   }
 
+  @Override
+  public InnReachResponseDTO borrowerRenew(String trackingId, String centralCode, BorrowerRenewDTO borrowerRenew) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var calculatedDueDate = transaction.getHold().getNeedBefore();
+    var requestedDueDate = borrowerRenew.getDueDateTime();
+
+    if (calculatedDueDate >= requestedDueDate) {
+      transaction.setState(BORROWER_RENEW);
+      transactionRepository.save(transaction);
+    } else {
+      try {
+        var recallItem = new InnReachRecallItemDTO().dueDateTime(calculatedDueDate);
+        transactionService.recallItem(trackingId, centralCode, recallItem);
```

Comment:
> Issue a D2IR recall request to the central server using the current due date of the loan

There is no need to call any local endpoint. The requirement means that you need to call an endpoint of central server (i.e. by using `InnReachExternalService`)

Reply:
thanks for help. Fixed